### PR TITLE
SO versions for cmake and libtool

### PR DESCRIPTION
**Pull request by cebe**
this updates the version-sync script to generate proper SO versions
which will be used by cmake and libtool to create version symlinks
on the system when a library is installed as well as setting the SO
version in the binary.

To see what this does, you have to configure tox with a prefix:

    ./configure --prefix=/tmp/tox-with-libtool
    mkdir cbuild && cd cbuild && cmake -DCMAKE_INSTALL_PREFIX=/tmp/tox-with-cmake ..

Then run `make && make install`.

in both instances you should see the following installed in `lib/`:

    libtoxcore.so -> libtoxcore.so.1.4.0
    libtoxcore.so.1 -> libtoxcore.so.1.4.0
    libtoxcore.so.1.4.0

inside the binary the soname should be the one with .1 and it should not
contain the full version:

    $ objdump -p libtoxcore.so.1.4.0 | grep SONAME
      SONAME               libtoxcore.so.1

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,15 +2,47 @@ cmake_minimum_required(VERSION 2.8.6)
...
-  "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
+set(PROJECT_VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
+
+# Update versions in various places
+# This must be run before setting SOVERSION, because SOVERSION is read from
+# a file generated by this script
+find_program(SHELL NAMES sh dash bash zsh)
+if(SHELL)
+  execute_process(
+      COMMAND ${SHELL} ${toxcore_SOURCE_DIR}/other/version-sync
```
Comment (jin-eld) [inline CMakeLists.txt]: I don't really get this part.. I thought version-sync is supposed to be run once before the release from the "outside" of any build, and automatically update configure.ac, cmake files and the so.version file?

**Review by jin-eld — COMMENTED** [review]
@cebe There's some weird stuff going on. If I run the version-sync script and then try the cmake build, it will revert the changes made by the version-sync script, that's really confusing; I guess it's related the piece of code that I commented on earlier, why would it be doing that?

File: CMakeLists.txt (hunk 1 above)
Comment (cebe) [reply CMakeLists.txt]: @iphydf can you please clarify why the version-sync script is run by cmake and not as a command on its own?

**Comment by cebe** [comment]
decoupled version-sync from cmake, ready to merge imo.

**Pull request merged**